subsys/stats/stats.c

```diff
@@ -221,6 +221,10 @@ stats_register(const char *name, struct stats_hdr *hdr)
 	struct stats_hdr *prev;
 	struct stats_hdr *cur;
 
+	if (strnlen(name, CONFIG_STATS_MAX_NAME_LEN) == CONFIG_STATS_MAX_NAME_LEN) {
+		return -ENAMETOOLONG;
+	}
```

Comment:
Not really sure on this? Seems like an MCUmgr limitation, don't see why it should apply to stats as a whole

Reply:
It's a sensible check on its own I'd say, but if we don't want to make any changes there, I could make a config in stat_mgmt/Kconfig instead, just for use when encoding.

Reply:
Stats system has been in zephyr since version 1.11 so would prefer not to make possible breaking changes to it at this point unless they're fixes